routes/api/users.js

```diff
@@ -26,4 +28,22 @@ router.post(
   validateUserLogin, userLogin
 );
 
+// signup or login with facebook
+router.get('/auth/facebook', facebookPassportRoutes.authenticate());
+
+// facebook callback route
+router.get('/auth/facebook/callback', facebookPassportRoutes.callback());
+
+// facebook success page
+router.get('/auth/facebook/success', (req, res) => res.status(200).json({
+  message: 'Successfully logged in with facebook!',
+  status: 200
+}));
+
+// facebook failure page
+router.get('/auth/facebook/failure', (req, res) => res.status(401).json({
+  message: 'Facebook login failed!',
+  status: 401
+}));
```

Comment:
Doesn't it make more sense to redirect the user the homepage on success(with this success message), and back to the login page on failure (with this failure message)? I reckon passport provides "Flash messages" also for this.
Why have you opted to create two new routes for success and failure?

Reply:
I think @Darthrighteous your comment will apply on the frontend.